x/clp/keeper/msg_server.go

```diff
@@ -998,3 +1001,30 @@ func (k msgServer) ModifyLiquidityProtectionRates(goCtx context.Context, msg *ty
 	ctx.EventManager().EmitEvents(events)
 	return response, nil
 }
+
+func (k msgServer) UpdateSwapFeeRate(goCtx context.Context, msg *types.MsgUpdateSwapFeeRateRequest) (*types.MsgUpdateSwapFeeRateResponse, error) {
+	response := &types.MsgUpdateSwapFeeRateResponse{}
+
+	// defensive programming
+	if msg == nil {
+		return response, errors.Errorf("msg was nil")
+	}
+
+	if err := msg.ValidateBasic(); err != nil {
```

Comment:
Just curious, is there any way you can think of, through which someone can reach this point of the code, without executing validate basic once already